### PR TITLE
UIIN-2459: Prepare new source values for shadow instances

### PR DESCRIPTION
**Pull request by OleksandrHladchenko1**
## Purpose

Per the ECS architects, new source values are needed to distinguish "shadow" instances in member library tenants from regular instances. To that end, FOLIO will add the following source values:

MARC-shared (UI value) = CONSORTIUM-MARC (BE value): used for shadows of Shared instances with Source = MARC
FOLIO-shared (UI value) = CONSORTIUM-FOLIO (BE value): used for shadows of Shared instances with Source = FOLIO

## Approach

* Added `SOURCE_VALUES` object that contains source values
* Added translations
* Created function `getSourceValue` that receives source value and returns 'FOLIO-shared' when BE value is `CONSORTIUM-FOLIO` and returns 'MARC-shared' when BE value is `CONSORTIUM-MARC`
* When instance is shared, hide `Edit instance` button

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/Instance/InstanceDetails/InstanceAdministrativeView/InstanceAdministrativeView.js [hunk 1]
```diff
@@ -10,7 +10,8 @@ import {
   Row,
   Col,
   KeyValue,
-  Highlighter
+  Highlighter,
+  NoValue
```
Comment (mariia-aloshyna) [inline src/Instance/InstanceDetails/InstanceAdministrativeView/InstanceAdministrativeView.js:14]:
```suggestion
  NoValue,
```

File: src/Instance/InstanceDetails/InstanceAdministrativeView/InstanceAdministrativeView.js [hunk 2]
```diff
@@ -27,7 +28,23 @@ import {
 } from '../../../components';
 
 import StatisticalCodesList from './StatisticalCodesList';
-import { QUERY_INDEXES } from '../../../constants';
+import { QUERY_INDEXES, SOURCE_VALUES } from '../../../constants';
```
Comment (mariia-aloshyna) [inline src/Instance/InstanceDetails/InstanceAdministrativeView/InstanceAdministrativeView.js:31]:
```suggestion
import {
  QUERY_INDEXES,
  SOURCE_VALUES,
} from '../../../constants';
```

File: src/Instance/InstanceDetails/InstanceAdministrativeView/InstanceAdministrativeView.js [hunk 3]
```diff
@@ -27,7 +28,23 @@ import {
...
+
+  if (source === SOURCE_VALUES.CONSORTIUM_FOLIO) {
+    return <FormattedMessage id="ui-inventory.folioShared" />;
+  }
+
+  if (source === SOURCE_VALUES.CONSORTIUM_MARC) {
+    return <FormattedMessage id="ui-inventory.marcShared" />;
+  }
+
+  return source;
```
Comment (mariia-aloshyna) [inline src/Instance/InstanceDetails/InstanceAdministrativeView/InstanceAdministrativeView.js:46]: Add `MARC` and `FOLIO` values to the SOURCE_VALUES, please
```suggestion
  return SOURCE_VALUES[source] || source;
```

File: src/ViewInstance.js [hunk 4]
```diff
@@ -228,6 +231,12 @@ class ViewInstance extends React.Component {
     this.props.mutator.allInstanceItems.reset();
   }
 
+  isMARCSource = (instance) => {
+    const instanceRecordsSource = instance?.source;
```
Comment (mariia-aloshyna) [inline src/ViewInstance.js:235]:
```suggestion
    const instanceRecordSource = instance?.source;
```

File: src/ViewInstance.js [hunk 5]
```diff
@@ -228,6 +231,12 @@ class ViewInstance extends React.Component {
     this.props.mutator.allInstanceItems.reset();
   }
 
+  isMARCSource = (instance) => {
```
Comment (mariia-aloshyna) [inline src/ViewInstance.js:234]: We have `isMARCSource` already in `utils`, can we use it?

File: src/ViewInstance.js [hunk 6]
```diff
@@ -479,6 +491,8 @@ class ViewInstance extends React.Component {
       };
     };
 
+    const isInstanceShared = isSourceConsortiumFolio || isSourceConsortiumMARC;
```
Comment (mariia-aloshyna) [inline src/ViewInstance.js:494]:
```suggestion
    const isInstanceShared = instance?.source.startWith(CONSORTIUM_PREFIX);
```

File: src/ViewInstance.js [hunk 7]
```diff
@@ -506,7 +520,7 @@ class ViewInstance extends React.Component {
       <>
         {showInventoryMenuSection && (
           <MenuSection label={intl.formatMessage({ id: 'ui-inventory.inventory.label' })} id="inventory-menu-section">
-            {canEditInstance && (
+            {canEditInstance && !isInstanceShared && (
```
Comment (mariia-aloshyna) [inline src/ViewInstance.js:523]: A central tenant won't have a source value started with CONSORTIUM- pref, but all instances should be shared for him and editing instance should be enabled

File: src/constants.js [hunk 8]
```diff
@@ -629,4 +629,10 @@ export const RECORD_SOURCE = {
   UC: 'UC',
 };
 
+export const SOURCE_VALUES = {
```
Comment (mariia-aloshyna) [inline src/constants.js:632]: Add `FOLIO`, please

**Review by mariia-aloshyna — COMMENTED** [review]
Also, cover the new functionality with unit tests, please

**Review by vashjs — APPROVED** [review]

File: src/ViewInstance.js [hunk 9]
```diff
@@ -505,7 +509,7 @@ class ViewInstance extends React.Component {
       <>
         {showInventoryMenuSection && (
           <MenuSection label={intl.formatMessage({ id: 'ui-inventory.inventory.label' })} id="inventory-menu-section">
-            {canEditInstance && (
+            {canEditInstance && !canMemberLibraryEditInstance && (
```
Comment (mariia-aloshyna) [inline src/ViewInstance.js:512]:
```suggestion
            {canEditInstance && !suppressEditInstanceForMemberTenant && (
```

File: src/ViewInstance.js [hunk 10]
```diff
@@ -478,6 +480,8 @@ class ViewInstance extends React.Component {
       };
     };
 
+    const canMemberLibraryEditInstance = instance?.source.startsWith(CONSORTIUM_PREFIX) && checkIfUserInMemberTenant(stripes);
```
Comment (mariia-aloshyna) [inline src/ViewInstance.js:483]:
```suggestion
    const suppressEditInstanceForMemberTenant = checkIfUserInMemberTenant(stripes) && instance?.source.startsWith(CONSORTIUM_PREFIX);
```

**Review by mariia-aloshyna — APPROVED** [review]

**Pull request merged**